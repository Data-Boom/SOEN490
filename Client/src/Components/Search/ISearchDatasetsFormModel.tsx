import * as Yup from 'yup'

export interface ISearchDatasetsFormModel {
  datasetId?: number[],
  material?: string[],
  firstName?: string,
  lastName?: string,
  year?: number,
  categoryId?: number,
  subcategoryId?: number
}

export interface ICategory {
  name: string,
  id: number
}
export interface ISubcategory {
  name: string,
  id: number
}

export const defaultSearchDatasetsModel: ISearchDatasetsFormModel = {
  datasetId: [],
  material: [],
  firstName: undefined,
  lastName: undefined,
  year: undefined,
  categoryId: undefined,
  subcategoryId: undefined
}

//todo complete Yup validation
export const searchDatasetsValidationSchema = Yup.object()