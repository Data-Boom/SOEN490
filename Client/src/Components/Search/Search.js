import { Box, Container, Divider, FormControl, FormControlLabel, Grid, InputLabel, Radio, RadioGroup, Select, Switch, Table, TableBody, TableCell, TableContainer, TableHead, TableRow, TextField } from '@material-ui/core';
import React, { useState } from 'react';

import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

export default function Search() {

  const [formValues, setFormValues] = useState({
    oxidizer: '',
    year: '',
    outputFormat: '',
    categories: '',
    subcategories: '',
    fuel: '',
    author: '',
    dilutent: '',
  })

  const [isCaseSensitive, setIsCaseSensitive] = useState(false)
  const [categories, setCategories] = useState([{ value: 1, text: "test1" }, { value: 2, text: "test2" }, { value: 3, text: "test3" }, { value: 4, text: "new value"}])
  const [rows, setRows] = useState([
    { title: "test" },
    { title: "test3" },
    { title: "test4" },
    { title: "test5" },
    { title: "test2" }
  ])

  const toggleIsCaseSensitive = () => {
    setIsCaseSensitive(!isCaseSensitive)
  }

  const handleInputChange = (event) => {
    const { name, value } = event.target
    setFormValues({ ...formValues, [name]: value})
  }

  const handleSubmit = () => {
    alert(JSON.stringify(formValues, null, 2) + `\n  ${isCaseSensitive}`)
  }

  const renderSearch = () => {
    console.log(JSON.stringify(formValues, null, 2))
    return (
      <>
        <Typography variant='h4' align="left">Search</Typography>
        <Grid container spacing={4}>
          <Grid item sm={3}>
            <TextField fullWidth label="Oxidizer" variant="outlined" name="oxidizer" value={formValues.oxidizer} onChange={handleInputChange} />
          </Grid>

          <Grid item sm={3}>
            <TextField fullWidth label="Year" variant="outlined" name="year" value={formValues.year} onChange={handleInputChange} />
          </Grid>

          <Grid item sm={2}>
            <FormControl variant="outlined" fullWidth>
              <InputLabel htmlFor="outlined-outFormat-native-simple" >Output format</InputLabel>
              <Select
                native
                label="categories"
                name="outputFormat" value={formValues.outputFormat} onChange={handleInputChange}
              >
                <option aria-label="None" value="" />
                {categories.map(option => <option value={option.value}>{option.text}</option>)}
              </Select>
            </FormControl>
          </Grid>

          <Grid item sm={2}>
            <FormControl variant="outlined" fullWidth>
              <InputLabel htmlFor="outlined-outFormat-native-simple" >Categories</InputLabel>
              <Select
                native
                label="categories"
                name="categories" value={formValues.categories} onChange={handleInputChange}
              >
                <option aria-label="None" value="" />
                {categories.map(option => <option value={option.value}>{option.text}</option>)}
              </Select>
            </FormControl>
          </Grid>

          <Grid item sm={2}>
            <Button variant="contained" color="primary" onClick={handleSubmit}>Search Database</Button>
          </Grid>
        </Grid>

        <Grid container spacing={4}>
          <Grid item sm={3}>
            <TextField label="Subcategories" variant="outlined" fullWidth name="subcategories" value={formValues.subcategories} onChange={handleInputChange} />
          </Grid>

          <Grid item sm={3}>
            <TextField label="Fuel" variant="outlined" fullWidth name="fuel" value={formValues.fuel} onChange={handleInputChange} />
          </Grid>
        </Grid>

        <Grid container spacing={4}>
          <Grid item sm={3}>
            <TextField label="Author" variant="outlined" fullWidth name="author" value={formValues.author} onChange={handleInputChange} />
          </Grid>

          <Grid item sm={3}>
            <TextField label="Diluent" variant="outlined" fullWidth name="dilutent" value={formValues.dilutent} onChange={handleInputChange} />
          </Grid>
        </Grid>

        <Grid container spacing={4}>
          <Grid item>
            <FormControl>
              <FormControlLabel
                label="Case Sensitive"
                labelPlacement="start"
                control={
                  <Switch
                    name="isCaseSensitive"
                    checked={isCaseSensitive}
                    onChange={toggleIsCaseSensitive}
                  />}
              />
            </FormControl>
          </Grid>
        </Grid>
      </>
    )
  }

  const renderResults = () => {
    return (
      <>
        <Typography variant='h4' align="left">Results</Typography>
        <TableContainer>
          <Table size={"small"}>
            <TableHead>
              <TableRow>
                <TableCell>Title</TableCell>
                <TableCell align="right">Oxidizer</TableCell>
                <TableCell align="right">Category</TableCell>
                <TableCell align="right">Subcategory</TableCell>
                <TableCell align="right">Fuel</TableCell>
                <TableCell align="right">Diluent</TableCell>
                <TableCell align="right">Author</TableCell>
                <TableCell align="right">Year</TableCell>
                <TableCell align="right">Output Format</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              {renderRows(rows)}
            </TableBody>
          </Table>
        </TableContainer>
      </>
    )
  }

  const renderRows = (rows) => {
    return rows.map((row) => (
      <TableRow key={row.name}>
        <TableCell component="th" scope="row">
          {row.name}
        </TableCell>
        <TableCell align="right">{row.title}</TableCell>
        <TableCell align="right">{row.oxidizer}</TableCell>
        <TableCell align="right">{row.category}</TableCell>
        <TableCell align="right">{row.subcategory}</TableCell>
        <TableCell align="right">{row.fuel}</TableCell>
        <TableCell align="right">{row.diluent}</TableCell>
        <TableCell align="right">{row.author}</TableCell>
        <TableCell align="right">{row.year}</TableCell>
        <TableCell align="right">{row.outputFormat}</TableCell>
      </TableRow>
    ))
  }

  console.log("rendered form")

  return (
    <Container>
      <Box pt={4}>
        {renderSearch()}
      </Box>
      <Box pt={4}>
        {renderResults()}
      </Box>
    </Container>
  )
}