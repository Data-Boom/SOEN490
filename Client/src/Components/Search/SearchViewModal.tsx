import { Box, Button, Grid, Modal, Paper } from "@material-ui/core"
import React, { useState } from 'react'

import CancelIcon from "@material-ui/icons/Cancel"
import { IDatasetModel } from "../../Models/Datasets/IDatasetModel"
import SearchView from "./SearchView"
import { classStyles } from "../../appTheme"

interface IProps {
  onDatasetsSelected: (foundDatasets: IDatasetModel[]) => void
}

export const SearchViewModal = (props: IProps) => {
  const { onDatasetsSelected } = { ...props }
  const [open, setOpen] = useState(false)

  const handleDatasetsSelected = (foundDatasets: IDatasetModel[]) => {
    onDatasetsSelected(foundDatasets)
    setOpen(false)
  }

  return (
    <>
      <Grid item>
        <Button id="add-dataset" onClick={() => setOpen(true)} color="primary" variant="contained">Add dataset To Graph</Button>
      </Grid>
      <Modal
        open={open}
        onClose={() => setOpen(false)}
        className={classStyles().modalsearch}
      >
        <Paper elevation={3}>
          <Box m={5}>
            <Grid container justify="flex-end">
              <Grid item>
                <CancelIcon color="primary" onClick={() => setOpen(false)} />
              </Grid>
            </Grid>
            <SearchView
              handleDatasetsSelected={handleDatasetsSelected}
            />
          </Box>
        </Paper>
      </Modal>
    </>
  )
}