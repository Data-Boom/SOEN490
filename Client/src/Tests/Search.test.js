import React from 'react';
import renderer from "react-test-renderer";
import { shallow } from 'enzyme';
import Search from '../Components/Search/Search'

describe('Search Component', () => {
    it("should match snapshot", () => {
        const tree = renderer.create(<Search />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});