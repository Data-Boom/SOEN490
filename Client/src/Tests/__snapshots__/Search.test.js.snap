// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should match snapshot 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg"
>
  <div
    className="MuiBox-root MuiBox-root-1"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 MuiTypography-alignLeft"
    >
      Search
    </h4>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-3"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
          >
            Oxidizer
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="oxidizer"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  Oxidizer
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-3"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
          >
            Year
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="year"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  Year
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="outlined-outFormat-native-simple"
          >
            Output format
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <select
              aria-invalid={false}
              className="MuiSelect-root MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              multiple={false}
              name="outputFormat"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              value=""
            >
              <option
                aria-label="None"
                value=""
              />
              <option
                value={1}
              >
                test1
              </option>
              <option
                value={2}
              >
                test2
              </option>
              <option
                value={3}
              >
                test3
              </option>
              <option
                value={4}
              >
                new value
              </option>
            </select>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  categories
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            htmlFor="outlined-outFormat-native-simple"
          >
            Categories
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <select
              aria-invalid={false}
              className="MuiSelect-root MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              multiple={false}
              name="categories"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              value=""
            >
              <option
                aria-label="None"
                value=""
              />
              <option
                value={1}
              >
                test1
              </option>
              <option
                value={2}
              >
                test2
              </option>
              <option
                value={3}
              >
                test3
              </option>
              <option
                value={4}
              >
                new value
              </option>
            </select>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  categories
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-2"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Search Database
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-3"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
          >
            Subcategories
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="subcategories"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  Subcategories
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-3"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
          >
            Fuel
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="fuel"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  Fuel
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-3"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
          >
            Author
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="author"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  Author
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-3"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
          >
            Diluent
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              name="dilutent"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  Diluent
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiFormControl-root"
        >
          <label
            className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementStart"
          >
            <span
              className="MuiSwitch-root"
            >
              <span
                aria-disabled={false}
                className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-6 MuiSwitch-switchBase MuiSwitch-colorSecondary"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={null}
              >
                <span
                  className="MuiIconButton-label"
                >
                  <input
                    checked={false}
                    className="PrivateSwitchBase-input-9 MuiSwitch-input"
                    disabled={false}
                    name="isCaseSensitive"
                    onChange={[Function]}
                    type="checkbox"
                  />
                  <span
                    className="MuiSwitch-thumb"
                  />
                </span>
              </span>
              <span
                className="MuiSwitch-track"
              />
            </span>
            <span
              className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
            >
              Case Sensitive
            </span>
          </label>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-10"
  >
    <h4
      className="MuiTypography-root MuiTypography-h4 MuiTypography-alignLeft"
    >
      Results
    </h4>
    <div
      className="MuiTableContainer-root"
    >
      <table
        className="MuiTable-root"
        role={null}
      >
        <thead
          className="MuiTableHead-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root MuiTableRow-head"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
              scope="col"
            >
              Title
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall"
              scope="col"
            >
              Oxidizer
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall"
              scope="col"
            >
              Category
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall"
              scope="col"
            >
              Subcategory
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall"
              scope="col"
            >
              Fuel
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall"
              scope="col"
            >
              Diluent
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall"
              scope="col"
            >
              Author
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall"
              scope="col"
            >
              Year
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall"
              scope="col"
            >
              Output Format
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root"
          role={null}
        >
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              role="cell"
              scope="row"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            >
              test
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
          </tr>
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              role="cell"
              scope="row"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            >
              test3
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
          </tr>
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              role="cell"
              scope="row"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            >
              test4
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
          </tr>
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              role="cell"
              scope="row"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            >
              test5
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
          </tr>
          <tr
            className="MuiTableRow-root"
            role={null}
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
              role="cell"
              scope="row"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            >
              test2
            </td>
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
            <td
              aria-sort={null}
              className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
